crates/wasi-http/tests/all/p3/incoming.rs

```diff
@@ -39,17 +39,17 @@ async fn run_wasi_http<E: Into<ErrorCode> + 'static>(
     wasmtime_wasi_http::p3::add_to_linker(&mut linker)?;
     let instance = linker.instantiate_async(&mut store, &component).await?;
     let proxy = Proxy::new(&mut store, &instance)?;
-    let handle = proxy.handle(&mut store, req).await?;
-    let res = match handle.get(&mut store).await? {
+    let handle = proxy.handle(&mut store, req);
+    let res = match instance.run(&mut store, handle).await?? {
```

Comment:
Assuming we go with my suggestion above, I'd expect this to look like:

```suggestion
    let handle = proxy.handle(&mut store, req)?;
    let res = match instance.run(&mut store, handle).await? {
```

so just like before

Reply:
Note that you still need the double question mark (`??`) since `Instance::run` returns a `Future<Output = Result<R>>` where `R` is the output type of the input future, which is also a `Result` in this case.  I could add a `Instance::run_flat` method which flattens the `Result<Result<_>>` into a `Result` if desired.  BTW, I sure wish https://doc.rust-lang.org/std/result/enum.Result.html#method.flatten could be stabilized.